src/main/java/me/magnet/consultant/Consultant.java

```diff
@@ -602,8 +604,9 @@ private void updateValidatedConfig(Properties newConfig) {
 
 	/**
 	 * Tears any outstanding resources down.
+	 * @throws InterruptedException If it got interrupted while waiting for any open tasks
 	 */
-	public void shutdown() {
+	public void shutdown() throws InterruptedException {
```

Comment:
Was wondering if we could do this without altering the method signature, but I don't think that's very desirable. This should force us to a new major release.